core/src/proto/types.rs

```diff
@@ -1169,6 +1224,18 @@ impl Tx {
         None
     }
 
+    /// Look-up the first memo field found in the transaction's
+    /// sections.
+    pub fn get_memo(&self) -> Option<&Memo> {
+        self.sections.iter().find_map(|sec| {
```

Comment:
It seems that transactions are slightly malleable: the transaction sections can be reordered without invalidating the signature sections (which only seem to test for the presence of certain section hashes). No? (cc @grarco ). If so, then we should either fix this malleability, or we should avoid operations like "get the first memo section in the transaction" (since the memo sections may have been reordered).